tinybmp/src/lib.rs

```diff
@@ -1,251 +1,243 @@
 
         let image_data = &bytes[header.image_data_start..];
 
-        Ok(BmpRaw { header, image_data })
+        Ok(Self {
+            header,
+            image_data,
+            color_type: PhantomData,
+        })
     }
 
-    /// Get a reference to the range of bytes that represents the pixel data in the image
-    pub fn image_data(&'a self) -> &'a [u8] {
-        self.image_data
+    /// Returns the BPP (bits per pixel) for this image.
+    pub fn color_bpp(&self) -> Bpp {
+        self.header.bpp
     }
 
-    /// Get the image width in pixels
-    pub fn width(&self) -> u32 {
-        self.header.image_width
-    }
-
-    /// Get the image height in pixels
-    pub fn height(&self) -> u32 {
-        self.header.image_height
-    }
-
-    /// Get image dimensions as `(width, height)` in pixels
-    pub fn dimensions(&self) -> (u32, u32) {
-        (self.header.image_width, self.header.image_height)
+    /// Returns a slice containing the raw image data.
+    pub fn raw_image_data(&self) -> &[u8] {
+        self.image_data
     }
 
-    /// Get the BPP (bits per pixel) for this image
-    pub fn bpp(&self) -> u32 {
-        u32::from(self.header.bpp)
+    /// Returns an iterator over the raw pixels in the image.
+    ///
+    /// The iterator returns the raw pixel colors as `u32` values. To automatically convert the raw
+    /// values into the color specified by `C` use [`pixels`] instead.
+    ///
+    /// [`pixels`]: #method.pixels
+    pub fn raw_pixels<'b>(&'b self) -> RawPixels<'b, 'a, C> {
+        RawPixels::new(self)
     }
 
     /// Returns the row length in bytes.
     ///
     /// Each row in a BMP file is a multiple of 4 bytes long.
     fn bytes_per_row(&self) -> usize {
-        let bits_per_row = self.width() as usize * self.bpp() as usize;
+        let bits_per_row = self.header.image_width as usize * usize::from(self.header.bpp.bits());
 
         (bits_per_row + 31) / 32 * (32 / 8)
     }
 }
 
-impl<'a> IntoIterator for &'a BmpRaw<'a> {
-    type Item = Pixel;
-    type IntoIter = BmpIterator<'a>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        let pixel_stride = match self.bpp() {
-            1 | 8 | 16 | 24 | 32 => self.bpp() as usize,
-            depth => panic!("Bit depth {} not supported", depth),
-        };
-
-        BmpIterator {
-            bmp: self,
-            pixel_data: self.image_data(),
-            pixel_stride,
-            x: 0,
-            y: 0,
-            bit_idx: 0,
-        }
+impl<'a> Bmp<'a, ()> {
+    /// Create a bitmap object from a byte slice.
+    ///
+    /// The created object keeps a shared reference to the input and does not dynamically allocate
+    /// memory.
+    ///
+    /// In contrast to the [`from_slice`] constructor no color type needs to be specified when
+    /// calling this method. This will disable all functions that requires a specified color type,
+    /// like the [`pixels`] method.
```

Comment:
Could we mention e-g by name here as well, something about using `from_slice` if using this crate with e-g.